homeworks/lesson_8_hw.js

```diff
@@ -0,0 +1,76 @@
+/* Створити базовий клас Employee з властивостями id, salary та статичним методом для порівняння властивостей salary, який
+  повертає строку з тесктом User with id ... has greater salary, або у випадку рівності значень
+  повертає salaries are equal. Створити два класи Dev та PM, які успадковують клас Employee та кожен
+  по своєму реалізує метод getJobType(), використовуючи для jobType приватну властивість. Кожен з класів 
+  повинен бути організований у окремому модулі, які лежать у папці modules. Приклад імпортів та викликів нижче.*/
+
+   import { Dev } from "./modules/dev.js";
+   import { PM } from "./modules/pm.js";
+   import { Employee } from "./modules/employee.js";
+   import { sortBySalary } from "./helpers/sorter.js";
+
+const dev = new Dev(1, 2000);
+const pm = new PM(2, 2500);
+const dev2 = new Dev(3, 2000);
+console.log(dev.getJobType()); // виведе I am a Dev
+console.log(pm.getJobType()); // виведе I am a PM
+console.log(Employee.compareSalary(dev, pm)); // виведе User with id 2 has greater salary
+console.log(Employee.compareSalary(dev, dev2)); // виведе salaries are equal
+
+/* На основі ієрархії, побудованої у попередньому завданні, створити об'єкт класу Dev, 
+   зробити властивість id незмінною та унеможливити її видалення. Строки для перевірки 
+   зміни чи видалення властивості можна коментувати для перевірки роботи скрипта у подальшому. */
+
+   const newDev = new Dev(1, 3000);
+   // ваш код
+  
+   // Закоментувала код, що видає помилку TypeError: Cannot assign to read only property 'id' of object '#<Dev>'
+   // бо запиняється на цьому виведення подальше
+  //  Object.defineProperty(newDev, 'id', {
+  //   value: 1,
```

Comment:
```suggestion
  //   value: newDev.id,
```
Так значення залишиться тим самим, що і було зпочатку